### ISSUE TITLE
bal --budget: maximum of empty list

### ISSUE THREAD

**Issue by glguy** [simonmichael/hledger#1122]
Internal exception escapes when computing budgets on things where there are no periodic transactions.

```
2019/11/1 Example
  Assets:Checking  1
  Income:Salary  -1
```

```
$ hledger -f no-roi.ledger bal --budget
Budget performance in 2019/11/01:

hledger: Prelude.maximum: empty list
```

**Comment by simonmichael** [comment]
This one seems to be fixed in master, though I don't remember it:
```
$ hledger-1.15 -f c.j bal --budget
Budget performance in 2019/11/01:

hledger-1.15: Prelude.maximum: empty list
$ hledger -f c.j bal --budget
Budget performance in 2019/11/01:

  ||    2019/11/01 
==++===============
--++---------------
  || 0             
```

**Comment by simonmichael** [comment]
Ah, fixed by @adept in e9d300be I think.

**Comment by adept** [comment]
Yep, I remember fixing this
